packages/x-components/tests/e2e/cucumber/recommendations/recommendations.spec.ts

```diff
@@ -1,22 +1,44 @@
 import { And, Given, Then } from 'cypress-cucumber-preprocessor/steps';
-import { InstallXOptions } from '../../../src/x-installer/x-installer/types';
+import { InstallXOptions } from '../../../../src/x-installer/x-installer/types';
+
+let config: InstallXOptions['xModules'];
 
 Given('following config: max items to store is {int}', (maxItemsToRequest: number) => {
-  const config: InstallXOptions['xModules'] = {
+  config = {
     recommendations: {
       config: {
         maxItemsToRequest
       }
     }
   };
-  cy.visit('/test/recommendations', {
+});
+
+And('suggestions response being mock {string}', (mockName: string) => {
+  cy.intercept('https://api.empathy.co/topRecommendations', req => {
+    import(`./stubs/recommendations.stub`)
+      .then(mock => {
+        return mock[mockName as keyof typeof mock];
+      })
+      .then(mockData => {
+        req.reply(mockData);
+      });
+  }).as('recommendationsApi');
```

Comment:
I think I was able to solve the issue you are describing above:

```suggestion
  cy.intercept('https://api.empathy.co/topRecommendations', async req => {
    const module = await import(`./stubs/recommendations.stub`);
    req.reply((module as any)[mockName]);
  }).as('recommendationsApi');
```

This is working fine for me.

This change implies changing  the `recommendations.feature` `mockName` values to the new one.

Reply:
The issue is that you cannot pass an argument to the import function and make it work as Guille said. What you've suggested do the same but await.

Reply:
It actually fixes an error when running `test:e2e:ci` so I'm going with this approach.